### PR TITLE
Implement gss_set_cred_option

### LINKED ISSUE #75: Implement  gss_set_cred_option()
Although libkrb5 should be able to handle a mechanism w/o this function implemented, it seem that, at least in some version, lack of the function causes a series of events that will end up with libkrb5 trying to dereference a gssntlmssp context as a union context and crashes.

Implementing, even an empty function would prevent the issue.

### PR DESCRIPTION
**Pull request by simo5**
Add also a way to use this function to set NEGOTIATE flags for both ISC and ASC.

Fixes #75

### CODE REVIEW

**Comment by filipnavara** [comment]
Sure, will do.

**Comment by filipnavara** [comment]
I tried it on Ubuntu 20.04 under WSL and unfortunately it still seems to go through the default `gssspi_set_cred_option` implementation and ignores `gssntlm_set_cred_option`. I am not sure how the mechanism glue is working in detail and what to check. `lldb` shows that the correct build of `gssntlmssp.so` was used and I can put breakpoints inside the new methods.

File: src/gss_spi.c [hunk 1]
```diff
@@ -452,3 +452,12 @@ OM_uint32 gssspi_mech_invoke(OM_uint32 *minor_status,
     return gssntlm_mech_invoke(minor_status, desired_mech, desired_object,
                                value);
 }
+
+OM_uint32 gss_set_cred_option(OM_uint32 *minor_status,
```
Comment (filipnavara) [inline src/gss_spi.c:456]: Presumably this should be `gssspi_set_cred_option`? (Note that the `cred_handle` has different level of pointer indirection on that method!)

https://github.com/krb5/krb5/blob/2b34a007461065e0cab4490dfe1ae5ddd10da67b/src/lib/gssapi/mechglue/g_initialize.c#L745

Comment (simo5) [reply src/gss_spi.c:456]: gah, yeah, my tests call the internal functions directly and did not catch this ...
I knew asking for your review would be fruitful :-D

Comment (simo5) [reply src/gss_spi.c]: Hopefully fixed now

Comment (filipnavara) [reply src/gss_spi.c]: > I knew asking for your review would be fruitful :-D

Always happy to find obscure bugs in other people's security/cryptography/authentication libraries.

> Hopefully fixed now

Yep, works great now.

**Pull request merged**